From 0bfe6232a40dba83487e07d2a7cec73eaa151591 Mon Sep 17 00:00:00 2001
From: Eric Huss <eric@huss.org>
Date: Tue, 19 Jul 2022 07:22:39 -0700
Subject: [PATCH] Move rust-analyzer to DUP_TOOLS

The rust-analyzer documentation includes the step `cargo install`
into `~/.cargo/bin` when building from source. Since this has a chance
of accidentally overriding the user's copy, move this to DUP_TOOLS.
---
 src/lib.rs        | 7 +++----
 tests/cli-misc.rs | 4 ++--
 2 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/src/lib.rs b/src/lib.rs
index 2cbbfb1177..6ba0f26e46 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -27,7 +27,6 @@ pub static TOOLS: &[&str] = &[
     "rust-gdb",
     "rust-gdbgui",
     "rls",
-    "rust-analyzer",
     "cargo-clippy",
     "clippy-driver",
     "cargo-miri",
@@ -36,7 +35,7 @@ pub static TOOLS: &[&str] = &[
 // Tools which are commonly installed by Cargo as well as rustup. We take a bit
 // more care with these to ensure we don't overwrite the user's previous
 // installation.
-pub static DUP_TOOLS: &[&str] = &["rustfmt", "cargo-fmt"];
+pub static DUP_TOOLS: &[&str] = &["rust-analyzer", "rustfmt", "cargo-fmt"];
 
 // If the given name is one of the tools we proxy.
 pub fn is_proxyable_tools(tool: &str) -> Result<()> {
@@ -112,8 +111,8 @@ mod tests {
             assert!(is_proxyable_tools(tool).is_ok());
         }
         let message = "unknown proxy name: 'unknown-tool'; valid proxy names are 'rustc', \
-        'rustdoc', 'cargo', 'rust-lldb', 'rust-gdb', 'rust-gdbgui', 'rls', 'rust-analyzer', \
-        'cargo-clippy', 'clippy-driver', 'cargo-miri', 'rustfmt', 'cargo-fmt'";
+        'rustdoc', 'cargo', 'rust-lldb', 'rust-gdb', 'rust-gdbgui', 'rls', \
+        'cargo-clippy', 'clippy-driver', 'cargo-miri', 'rust-analyzer', 'rustfmt', 'cargo-fmt'";
         assert_eq!(
             is_proxyable_tools("unknown-tool").unwrap_err().to_string(),
             message
diff --git a/tests/cli-misc.rs b/tests/cli-misc.rs
index d7c8243614..08fe9ac6d6 100644
--- a/tests/cli-misc.rs
+++ b/tests/cli-misc.rs
@@ -413,8 +413,8 @@ fn rustup_failed_path_search() {
             broken,
             "unknown proxy name: 'fake_proxy'; valid proxy names are \
              'rustc', 'rustdoc', 'cargo', 'rust-lldb', 'rust-gdb', 'rust-gdbgui', \
-             'rls', 'rust-analyzer', 'cargo-clippy', 'clippy-driver', 'cargo-miri', \
-             'rustfmt', 'cargo-fmt'",
+             'rls', 'cargo-clippy', 'clippy-driver', 'cargo-miri', \
+             'rust-analyzer', 'rustfmt', 'cargo-fmt'",
         );
 
         // Hardlink will be automatically cleaned up by test setup code
